torchchat/export.py

```diff
@@ -28,6 +28,31 @@
 default_device = "cpu"
 
 
+"""
+Export Snapshot
+"""
+
+
+def export_snapshot(
+    model: nn.Module,
+    device: Optional[str] = None,
+    output_path: str = "model-snapshot.tc",
+) -> str:
+    """
+    Export the model as snapshot.
+
+    Args:
+        model: The model to be exported.
+        device: The device to run the model on.
+        output_path: The path to save the exported model.
+    Returns:
+        The path to the exported model.
+    """
+    assert output_path.endswith(".tc"), "use .tc extension for snapshots"
+    torch.save(model, output_path)
```

Comment:
Do we need the whole model or can we get away with just the state_dict? https://github.com/pytorch/torchchat/pull/1280/files

That said if we go with the slimmer state_dict, that's dependent on migration to the AO quant that supports this saving

Reply:
You rewrite the code as part of the quantization.  If you don't save the code, then you must exactly replicate what quantization options you used, create an empty model, quantize it, and then load the state dict over it.  Either you transfer the whole responsibility on the user (good luck, you'll die the death of many cuts when users make mistakes and complain that this facility does not work), or you need to save an ungodly amount of information about options used for the original quantization process.

Reply:
We can update the serialization logic as the AO migration finishes (i.e. this PR is good), but I'm not sure that's the case anymore with AO. I was under the impression that the model itself is unaffected and that only the weights are changed

https://github.com/pytorch/ao/blob/48fdd310b3977a0db2ceba37a7725192cd2aafd4/docs/source/serialization.rst#L62 

cc: @vmpuri @HDCharles

Reply:
you should be able to get by with just the state dict, there are a few apis that don't work that way, but all the subclass ones do. Thats like 75% of the reason that we went with subclasses instead of module swaps.

jack, you're link is a good resource, an alternate reference are our serialization tests to see what is explicitly tests

see e.g. https://github.com/pytorch/ao/blob/48fdd310b3977a0db2ceba37a7725192cd2aafd4/test/integration/test_integration.py#L1322-L1334